static/src/components/TaskView.vue

```diff
@@ -0,0 +1,85 @@
+<script lang="ts">
+import { defineComponent } from "vue";
+import ImageDisplay from "@/components/ImageDisplay.vue";
+import AnnotationButton from "@/components/AnnotationButton.vue";
+import ProofOfCondition from "@/components/ProofOfCondition.vue";
+import UserInformationLabel from "@/components/UserInformationLabel.vue";
+import { TasksService as API } from "@/api/services/TasksService";
+import { paramCase } from "change-case";
+export default defineComponent({
+  components: {
+    ImageDisplay,
+    AnnotationButton,
+    ProofOfCondition,
+    UserInformationLabel,
+  },
+
+  props: {
+    taskId: { type: String, required: true },
+    competency: { type: String, required: true },
+    user: { type: String, required: true },
+  },
+
+  data() {
+    return {
+      labels: [] as string[],
+      visible: true,
+      isAttentive: false,
+      isTrained: false,
+      imageId: undefined as string | void,
+    };
+  },
+
+  mounted() {
+    this.fetchLabels();
+    this.nextImage();
+  },
+
+  methods: {
+    async fetchLabels() {
+      let task = await API.getTask(this.taskId);
+      this.labels = task.labels;
+    },
+
+    async nextImage() {
+      this.imageId = await API.getNextImage(this.taskId).catch((e) => {
+        console.log(e);
+        this.imageId = undefined;
+      });
+    },
+
+    paramCase,
+  },
+});
+</script>
+
+<template>
+  <i-layout-content>
+    <ProofOfCondition
+      v-model:isAttentive="isAttentive"
+      v-model:isTrained="isTrained"
+    />
+    <UserInformationLabel :username="user" />
+    <div v-if="imageId">
+      <ImageDisplay id="image-display" :task-id="taskId" :image-id="imageId" />
+      <i-button-group block>
+        <AnnotationButton
+          v-for="label in labels"
+          :id="'annotation-button-' + paramCase(label)"
+          :key="label"
+          :label="label"
+          username="user"
```

Comment:
I dont know if you changed that because the diff is scuffed, but this should be `:username="user"`, right?

Reply:
It's the same on dev, but really does not look correct :thinking:

Reply:
Generally, I think we should rather emit an event when the button was pressed and let the task view execute the request itself (although this may rather be put into a separate PR?). What do you think?

Reply:
Yeah that would also save some of the attribute-passing to the button. Good Idea for a separate PR